app/controllers/api/v1/search_controller.rb

```diff
@@ -762,6 +794,26 @@ def self.find_matching_filters(facet:, filter_values:)
         matching_filters
       end
 
+      # take a term and match to a possible search facet/filter
+      # will return a single hash with keys as facet names, and values as an array of filter matches
+      def self.match_facet_filters_from_terms(term_list)
+        terms = term_list || []
+        filters_by_facet = {}
+        terms.each do |term|
+          facets = SearchFacet.find_facets_from_term(term)
```

Comment:
I suspect this operation is relatively cheap, but it's worth noting that we'll duplicate this operation in `format_facet_query_from_keyword` in hca_azul_client, so it might be nice if we could consolidate